internal/mesh/internal/controllers/sidecarproxy/fetcher/data_fetcher.go

```diff
@@ -192,20 +107,24 @@ func (f *Fetcher) FetchExplicitDestinationsData(
 
 		// As Destinations resource contains a list of destinations,
 		// we need to find the one that references our service and port.
-		d.Explicit = findDestination(dest.ServiceRef, dest.Port, us.Destinations)
+		d.Explicit = findDestination(dest.ServiceRef, dest.Port, us.Data)
```

Comment:
likely need to `continue` if `d.Explicit` returns `nil` due to cache inconsistency.